internal/errors/errors.go

```diff
@@ -0,0 +1,145 @@
+	ErrorInvalidSigner = errorsmod.Register(codespace, 24, "tx intended signer does not match the given signer")
+
+	// ErrorInvalidGasAdjustment defines an error for an invalid gas adjustment
+	ErrorInvalidGasAdjustment = errorsmod.Register(codespace, 25, "invalid gas adjustment")
+
+	// ErrInvalidHeight defines an error for an invalid height
+	ErrInvalidHeight = errorsmod.Register(codespace, 26, "invalid height")
+
+	// ErrInvalidVersion defines a general error for an invalid version
+	ErrInvalidVersion = errorsmod.Register(codespace, 27, "invalid version")
+
+	// ErrInvalidChainID defines an error when the chain-id is invalid.
+	ErrInvalidChainID = errorsmod.Register(codespace, 28, "invalid chain-id")
+
+	// ErrInvalidType defines an error an invalid type.
+	ErrInvalidType = errorsmod.Register(codespace, 29, "invalid type")
+
+	// ErrTxTimeoutHeight defines an error for when a tx is rejected out due to an
+	// explicitly set timeout height.
+	ErrTxTimeoutHeight = errorsmod.Register(codespace, 30, "tx timeout height")
+
+	// ErrUnknownExtensionOptions defines an error for unknown extension options.
+	ErrUnknownExtensionOptions = errorsmod.Register(codespace, 31, "unknown extension options")
+
+	// ErrWrongSequence defines an error where the account sequence defined in
+	// the signer info doesn't match the account's actual sequence number.
+	ErrWrongSequence = errorsmod.Register(codespace, 32, "incorrect account sequence")
+
+	// ErrPackAny defines an error when packing a protobuf message to Any fails.
+	ErrPackAny = errorsmod.Register(codespace, 33, "failed packing protobuf message to Any")
+
+	// ErrUnpackAny defines an error when unpacking a protobuf message from Any fails.
+	ErrUnpackAny = errorsmod.Register(codespace, 34, "failed unpacking protobuf message from Any")
+
+	// ErrLogic defines an internal logic error, e.g. an invariant or assertion
+	// that is violated. It is a programmer error, not a user-facing error.
+	ErrLogic = errorsmod.Register(codespace, 35, "internal logic error")
+
+	// ErrConflict defines a conflict error, e.g. when two goroutines try to access
+	// the same resource and one of them fails.
+	ErrConflict = errorsmod.Register(codespace, 36, "conflict")
+
+	// ErrNotSupported is returned when we call a branch of a code which is currently not
+	// supported.
+	ErrNotSupported = errorsmod.Register(codespace, 37, "feature not supported")
+
+	// ErrNotFound defines an error when requested entity doesn't exist in the state.
+	ErrNotFound = errorsmod.Register(codespace, 38, "not found")
+
+	// ErrIO should be used to wrap internal errors caused by external operation.
+	// Examples: not DB domain error, file writing etc...
+	ErrIO = errorsmod.Register(codespace, 39, "Internal IO error")
+
+	// ErrAppConfig defines an error occurred if min-gas-prices field in BaseConfig is empty.
+	ErrAppConfig = errorsmod.Register(codespace, 40, "error in app.toml")
```

Comment:
Should we consider pruning down the errors that aren't applicable to us? I don't think we need to copy over all of them, but I don't have a strong opinion on the matter. I think the idea is that these are now our errors

Reply:
yeah I was thinking we could do that. I'll try and remove all the ones we don't use anywhere in the codebase.

Reply:
question: Will the error code need to remain the same or can it change? Will changing the error code potentially be state machine breaking?

Reply:
I'm thinking specifically about [this](https://github.com/cosmos/ibc-go/blob/b40dbc646c7191aaee12d3a70243f9b6f3f80919/modules/core/04-channel/types/acknowledgement.go#L35-L41) function.

Reply:
It can change. Error codes are state machine breaking, but we don't plan to backport this to a minor release? If included in a major release, it won't be an issue. It does not need to maintain parity with the SDK errors. I believe just changing the codespace will change the ABCI error code anyways